project/plugins.sbt

```diff
@@ -1,10 +1,13 @@
-addSbtPlugin("org.scala-lang.modules" % "sbt-scala-module" % "1.0.14")
-
 if (System.getProperty("java.version").startsWith("1."))
   Seq()
 else
   // override to version that works on Java 9,
   // see https://github.com/scala/sbt-scala-module/issues/35
   Seq(addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.9.3"))
 
-addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.22")
+val scalaJSVersion =
+  Option(System.getenv("SCALAJS_VERSION")).filter(_.nonEmpty).getOrElse("0.6.23")
+
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % scalaJSVersion)
+addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "0.4.0")
```

Comment:
What's this about?

```
[warn] Found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[warn] 
[warn] 	* org.scala-js:sbt-scalajs:0.6.23 is selected over 0.6.22
[warn] 	    +- default:scala-xml-build:0.1-SNAPSHOT (scalaVersion=2.10, sbtVersion=0.13) (depends on 0.6.23)
[warn] 	    +- org.portable-scala:sbt-scalajs-crossproject:0.4.0 (scalaVersion=2.10, sbtVersion=0.13) (depends on 0.6.22)
[warn] 
[warn] 	* org.portable-scala:sbt-platform-deps:1.0.0 is selected over 1.0.0-M2
[warn] 	    +- org.scala-js:sbt-scalajs:0.6.23 (scalaVersion=2.10, sbtVersion=0.13) (depends on 1.0.0)
[warn] 	    +- org.portable-scala:sbt-crossproject:0.4.0 (scalaVersion=2.10, sbtVersion=0.13) (depends on 1.0.0-M2)
[warn] 	    +- org.portable-scala:sbt-scalajs-crossproject:0.4.0 (scalaVersion=2.10, sbtVersion=0.13) (depends on 1.0.0-M2)
[warn] 
[warn] Run 'evicted' to see detailed eviction warnings
```

Why doesn't the `sbt-scalajs-crossproject` plugin just bring in the corresponding `sbt-scalajs` core plugin?

Reply:
`sbt-scalajs-crossproject` is not only about  scala-js, but also native. I don't know it very well, but the docs say that the js plugin should be added (https://github.com/portable-scala/sbt-crossproject). It would be nice if this warning didn't show up though..